### PR TITLE
Fix broken invalid url handling on iOS 17

### PR DESCRIPTION
**Pull request by ajaysubra**
# Description

In iOS 17 apple changes how url's string initializer works and this was causing one of our tests that validates invalid urls to fail. I reverted the URL initializer to not use the new decoding and sticking with the pre iOS 17 logic of keeping invalid characters from decoding. This is fine since the string we are using the URL initializer is a static one and the idea is to align the code with the tests we had. 

[Change](https://ihor.pro/handling-a-new-url-parsing-behavior-in-ios-17-c08598d28659) in iOS 17 for more context.

# Check List

- [ ] Are you changing anything with the public API?
- [ ] Have you tested this change on real device?
- [ ] Are your changes backwards compatible with previous SDK Versions?
- [ ] Have you added unit test coverage for your changes?
- [ ] Have you verified that your changes are compatible with all the operating system version this SDK currently supports?

# Manual Test Plan

1.

# Supporting Materials

### CODE REVIEW

File: Sources/KlaviyoSwift/KlaviyoAPI.swift [hunk 1]
```diff
@@ -103,7 +103,11 @@ extension KlaviyoAPI.KlaviyoRequest {
     var url: URL? {
         switch endpoint {
         case .createProfile, .createEvent, .registerPushToken, .unregisterPushToken:
-            return URL(string: "\(environment.analytics.apiURL)/\(path)/?company_id=\(apiKey)")
+            if #available(iOS 17.0, *) {
+                return URL(string: "\(environment.analytics.apiURL)/\(path)/?company_id=\(apiKey)", encodingInvalidCharacters: false)
```
Comment (ndurell) [inline Sources/KlaviyoSwift/KlaviyoAPI.swift:107]: Nice

**Review by ndurell — APPROVED** [review]

File: .github/workflows/swift.yml [hunk 2]
```diff
@@ -19,7 +19,7 @@ jobs:
     runs-on: macos-13
     strategy:
       matrix:
-        xcode: ['14.3']
+        xcode: ['15.1']
```
Comment (kennyklaviyo) [inline .github/workflows/swift.yml:22]: is this change related?

Comment (evan-masseau) [reply .github/workflows/swift.yml:22]: Yeah, since the breaking code is related to a new swift version, he had to update the runner XCode version to access that new swift version.

**Review by evan-masseau — APPROVED** [review]

**Comment by evan-masseau** [comment]
@ajaysubra assume you won't mind, i updated this against master so we can see if CI 100% is fixed now.

**Comment by ajaysubra** [comment]
> hm CI is hung on 14.3 tests, which I think you removed. Might have to force-merge this one, maybe those tests are cached or something?

Thanks for updating with master. You are probably right that GH is caching something. Manually triggered the jobs 🤞🏽

**Review by kennyklaviyo — APPROVED** [review]

**Comment by ajaysubra** [comment]
Actions is passing but it's not updating the PR so I'm going to just merge this bypassing the checks.
image
